### PR TITLE
[android] Refactor API and Intent processing

### LINKED ISSUE #1128: Do not show "Enable location services" on every launch
Some users do not enable Location services at all and always see the "Please enable Location Services" annoying dialog on every app launch. The dialog appears anyway if the user presses the "my location" button in the app, so for better UX it makes sense to show dialog only once and store this fact in the settings, but reset this setting again if the location was successfully acquired (e.g. user enabled location services).

The main idea is to show this warning only once, if the user does not enable location services, and show it again if services were enabled, and then disabled again.

### LINKED ISSUE #3441: [android] Position arrow overlaps with bottom bar in navigation mode.
Samsung Galaxy S20 FE (Snapdragon), latest beta:

Upd: looks like it happens only when device is locked

photo_2022-09-18 12 08 19

### PR DESCRIPTION
**Pull request by rtsisyk**
- Remove MapTask abstraction in favor of Intents.
- Process intents and API after creation of Drape.
- Restore routing only after creation of Drape (fixes #3441).
- Start location only after creation of Drape (fixes #1128).
- Remove double parsing of `backurl` in URLs.

**Test Plan:**

- Check URLs from http://omaps.app/test.html.
- Check KML/KMZ/GPX files.
- Check "Pick point" from https://github.com/organicmaps/api-android.
- Download Maps -> any downloaded map -> Show on the map.
- Bookmarks -> any list -> click on a blue button the bottom right.
- Bookmarks -> any list -> click on any bookmark.
- Start navigation -> kill the app **via system settings** ->
  start the app again and check that routing planned is started.

### CODE REVIEW

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 14
Tested with all links from omaps.app/api
Back button in place page is good idea, but IMO this button can create incomprehension, because button leave app not place page.
What about AA?

File: android/app/src/main/java/app/organicmaps/MwmActivity.java [hunk 1]
```diff
@@ -979,57 +1015,23 @@ private void onIsolinesStateChanged(@NonNull IsolinesState type)
   @Override
   protected void onNewIntent(Intent intent)
   {
-    super.onNewIntent(intent);
+    // {@link see BaseMwmFragmentActivity.onCreate()}
+    final Intent initialIntent = intent.getParcelableExtra(SplashActivity.EXTRA_INITIAL_INTENT, Intent.class);
```
Comment (Jean-BaptisteC) [inline android/app/src/main/java/app/organicmaps/MwmActivity.java:1019]: Android Lint tell this is only available with API 33 and more

**Comment by rtsisyk** [comment]
> Back button in place page is good idea, but IMO this button can create incomprehension, because button leave app not place page.What is your main idea?

It should present only in the case of API calls that expect results to be returned to the caller. I had no intention to change that semantics.

**Comment by Jean-BaptisteC** [comment]
Back button is always present in place page even I close app after use Intent :)

**Comment by rtsisyk** [comment]
> Back button is always present in place page even I close app after use Intent :)

Yeah, a good catch!

```
     final boolean showBackButton = (intent != null && (Factory.isStartedForApiResult(intent) ||
-        TextUtils.isEmpty(Framework.nativeGetParsedBackUrl())));
+        !TextUtils.isEmpty(Framework.nativeGetParsedBackUrl())));
     mMapObject = mapObject;
     if (mapObject != null)
```

Fixed.

File: android/app/src/main/java/app/organicmaps/MwmActivity.java (hunk 1 above)
Comment (rtsisyk) [reply android/app/src/main/java/app/organicmaps/MwmActivity.java]: Fixed by using IntentCompat.getParcelableExtra() in all places across the project.

**Comment by Jean-BaptisteC** [comment]
Now, back button is never visible :)

**Comment by rtsisyk** [comment]
> Now, back button is never visible in place page :)

Try pick_point example from https://github.com/organicmaps/api-android

**Comment by Jean-BaptisteC** [comment]
Pick point works (return latitude, longitude)
Back button is not in place page when I used link from example website

**Review by biodranik — APPROVED** [review]
I'm ok with this change if API works.

**Comment by rtsisyk** [comment]
The back button in the place page is shown only in two cases:

 - "backurl" in om://map link is not empty
 - PICK_POINT=true from API example.

> Pick point works (return latitude, longitude) Back button is not in place page when I used link from example website

Try adding `backurl=`:

`adb shell am start -a android.intent.action.VIEW -d '"om://map?v=1&ll=22.17319,-159.65687&n=Kalalau%20Camping&ll=22.17168,-159.66096&n=Dream%20Beach&ll=22.17182,-159.65776&backurl=https%3A%2F%2Forganicmaps.app"'`

**Comment by rtsisyk** [comment]
> I'm ok with this change if API works.

Yes, API works better than before. This patch fixes race conditions on the start when routing, API, location, KML/KMZ/GPX were started before drape initialization and caused inconsistency. This PR fixes #1128 and should fix a hard-to-reproduce race conditions in #3441.

**Pull request merged**